File: 2249-CountTheHiddenSequences/2249-CountTheHiddenSequences.cpp
// Last updated: 11/05/2026, 01:31:22
class Solution {
public:
    int numberOfArrays(vector<int>& differences, int lower, int upper) {
        int low = lower, up = upper;
        for (int x:differences){
            low = max(lower,low+x);
            up = min(upper,up+x);
            if (low > upper || up < lower)
            return 0;
        }
        return up-low+1;
//         lo, up = lower, upper
//         for diff in differences:
//             lo, up = max(lower, lo + diff), min(upper, up + diff)
//             if lo > upper or up < lower: return 0

//         return up - lo + 1
    }
};

File: 2249-CountTheHiddenSequences/2249-CountTheHiddenSequences.py
# Last updated: 11/05/2026, 01:31:27
class Solution(object):
    def numberOfArrays(self, differences, lower, upper):
        """
        :type differences: List[int]
        :type lower: int
        :type upper: int
        :rtype: int
        """
        lo, up = lower, upper
        for diff in differences:
            lo, up = max(lower, lo + diff), min(upper, up + diff)
            if lo > upper or up < lower: return 0

        return up - lo + 1
        